policy-controller/runtime/src/args.rs

```diff
@@ -461,62 +451,6 @@ async fn grpc(
     Ok(())
 }
 
-async fn init_lease(client: Client, ns: &str, deployment_name: &str) -> Result<LeaseManager> {
-    // Fetch the policy-controller deployment so that we can use it as an owner
-    // reference of the Lease.
-    let api = k8s::Api::<Deployment>::namespaced(client.clone(), ns);
-    let deployment = api.get(deployment_name).await?;
-
-    let api = k8s::Api::namespaced(client, ns);
-    let params = PatchParams {
-        field_manager: Some("policy-controller".to_string()),
-        ..Default::default()
-    };
-    match api
-        .patch(
-            LEASE_NAME,
-            &params,
-            &kube::api::Patch::Apply(coordv1::Lease {
-                metadata: ObjectMeta {
-                    name: Some(LEASE_NAME.to_string()),
-                    namespace: Some(ns.to_string()),
-                    // Specifying a resource version of "0" means that we will
-                    // only create the Lease if it does not already exist.
-                    resource_version: Some("0".to_string()),
-                    owner_references: Some(vec![deployment.controller_owner_ref(&()).unwrap()]),
-                    labels: Some(
-                        [
-                            (
-                                "linkerd.io/control-plane-component".to_string(),
-                                "destination".to_string(),
-                            ),
-                            ("linkerd.io/control-plane-ns".to_string(), ns.to_string()),
-                        ]
-                        .into_iter()
-                        .collect(),
-                    ),
-                    ..Default::default()
-                },
-                spec: None,
-            }),
-        )
-        .await
-    {
-        Ok(lease) => tracing::info!(?lease, "Created Lease resource"),
-        Err(k8s::Error::Api(_)) => tracing::debug!("Lease already exists, no need to create it"),
-        Err(error) => {
-            tracing::error!(%error, "Failed to create Lease resource");
-            return Err(error.into());
```

Comment:
The only functional change in this diff is the removal of this log line. We should _either_ emit an error log _or_ return an error. In this case, returning the error means it will bubble up to a higher level error handler (ultimately being print on stderr).